lib/3scale/backend/transactor.rb

```diff
@@ -137,9 +143,17 @@ def do_authrep(method, provider_key, params, context_info)
 
         usage = params[:usage]
 
-        if (usage || params[:log]) && status.authorized?
+        filtered_usage = usage&.reject { |_metric, delta| delta.to_s == '0' }
```

Comment:
I think we can split the helper method to filter 0's so that you can use it for both report and authrep?

Reply:
Done. Check new commit.